x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/filter_agg/components/editor_form.tsx

```diff
@@ -36,6 +38,17 @@ export const FilterEditorForm: FilterAggConfigEditor['aggTypeConfig']['FilterAgg
         }}
         value={config || ''}
       />
+      {isValid === false ? (
+        <>
+          <EuiSpacer size="m" />
+          <EuiCallOut color="danger" iconType="alert" size="s">
+            <FormattedMessage
+              id="xpack.transform.agg.filterEditorForm.jsonInvalidErrorMessage"
+              defaultMessage="JSON is invalid."
+            />
+          </EuiCallOut>
+        </>
+      ) : undefined}
```

Comment:
according to [react docs](https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering), we should return `null` to prevent rendering

```suggestion
      ) : null}
```

Reply:
Updated here https://github.com/elastic/kibana/pull/137773/commits/6cde9458bb3ef80020abbb7b3024bfa929a165bc